### PR TITLE
Add monitor tests

### PR DESCRIPTION
**Pull request by jwhite3**
This PR adds tests for a "test" monitor instance. Mostly they just ensure that attributes are set correctly when the various methods are called and that no errors occur in generating the basic plots.

Also fixes a minor bug in the basic scatter plot generation.

### CODE REVIEW

File: tests/test_monitor.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+NAME = {'name': 'Test Monitor Name'}
+SUBPLOTS = {'subplots': True, 'subplot_layout': (2, 1)}
+LABELS = {'labels': ['a']}
+SIMPLE_SCATTER = {'x': 'b', 'y': 'c', 'plottype': 'scatter'}
+SIMPLE_LINE = {'x': 'b', 'y': 'c', 'plottype': 'line'}
+SIMPLE_IMAGE = {'x': 'b', 'y': 'c', 'z': 'd', 'plottype': 'image'}
+
+
+@pytest.fixture
+def datamodel_test_instance():
```
Comment (bourque) [inline tests/test_monitor.py:29]: Perhaps a matter of preference here, but I usually treat test functions like any other python function in regards to docstrings.  Might be a good idea to add one here (and in the other functions/classes/methods) to describe what the function does.

Comment (bourque) [reply tests/test_monitor.py:29]: To that end, it might be also useful to include a module level docstring

Comment (jwhite3) [reply tests/test_monitor.py:29]: Yep, good call. I just forgot to add comments and docstrings... 😞

File: tests/test_datamodel.py [hunk 2]
```diff
@@ -72,39 +83,50 @@ def test_ingest_format(self, datamodel_test_instance):
...
-        query = list(datamodel_test_instance.model.select().dicts())
+        """Test that the peewee model object is created successfully (post-ingestion)."""
+        datamodel_test_instance.ingest()  # The model is created via introspection. There must be a table with data.
 
+        # Check that a peewee model can be constructed and queried
+        query = list(datamodel_test_instance.model.select().dicts())
         assert len(query) == 3
 
+        # Check that the model columns are correct
+        assert list(datamodel_test_instance.model._meta.columns.keys()) == ['a', 'b', 'c']
```
Comment (robelgeda) [inline tests/test_datamodel.py]: Is the output of the list generation sorted? Might be worthwhile given the short length.

File: tests/test_monitor.py [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+    """Test class for testing the monitorframe BaseMonitor."""
+    def test_monior_init(self, monitor_test_instance):
+        """Test that attributes are created or set correctly for each configuration."""
+        assert monitor_test_instance.model.new_data is not None  # New data should be retrieved by default
+
+        # Check that a table object is created and that the name is set correctly
+        assert (
+                monitor_test_instance._table is not None and
+                monitor_test_instance._table._meta.table_name == monitor_test_instance.__class__.__name__
+        )
```
Comment (robelgeda) [inline tests/test_monitor.py:101]: for these I like to add a printout message. This helps with Travis too. For example:

`assert a == b, "a which is {} is not equal to b which is {}".fromat(a, b)`

That way you can say what you tested for and print out any relevant information when the assertion fails. Not sure how useful this is here though.

**Review by robelgeda — APPROVED** [review]
Looks good to me. I added some comments but they are minor suggestions. No bugs detected.

File: tests/test_datamodel.py (hunk 2 above)
Comment (jwhite3) [reply tests/test_datamodel.py]: @robelgeda I believe that the behavior for pandas data frames is to preserve the order of the input data, but you're right, there's no guarantee that this behavior will persist across new versions of pandas. Good call!

File: tests/test_monitor.py (hunk 3 above)
Comment (jwhite3) [reply tests/test_monitor.py:101]: @robelgeda Turns out that this is redundant behavior if you're using `pytest`. In the default reports for `pytest` if the comparison fails there will be a line under the test that fails that "explains" why the assertion failed (with values included)

File: tests/test_datamodel.py (hunk 2 above)
Comment (jwhite3) [inline tests/test_datamodel.py]:
```suggestion
        assert sorted(list(datamodel_test_instance.model._meta.columns.keys())) == ['a', 'b', 'c']
```

**Pull request merged**